java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java

```diff
@@ -0,0 +1,89 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.stream;
+
+import glide.api.commands.StreamBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Optional arguments for {@link StreamBaseCommands#xgroupCreate(String, String, String,
+ * StreamGroupOptions)}
+ *
+ * @see <a href="https://valkey.io/commands/xgroup-create/">valkey.io</a>
+ */
+public final class StreamGroupOptions {
+
+    public static final String MAKE_STREAM_REDIS_API = "MKSTREAM";
+    public static final String ENTRIES_READ_REDIS_API = "ENTRIESREAD";
+
+    /** If the stream doesn't exist, creates a new stream with a length of <code>0</code>. */
+    boolean makeStream;
+
+    /**
+     * An arbitrary ID (that isn't the first ID, last ID, or the zero <code>"0-0"</code>. Use it to
+     * find out how many entries are between the arbitrary ID (excluding it) and the stream's last
+     * entry.
+     *
+     * @since Redis 7.0.0
+     */
+    String entriesRead;
+
+    /**
+     * Options for {@link StreamBaseCommands#xgroupCreate(String, String, String, StreamGroupOptions)}
+     *
+     * @param makeStream If the stream doesn't exist, creates a new stream with a length of <code>0
+     *     </code>.
+     */
+    public StreamGroupOptions(Boolean makeStream) {
```

Comment:
I'm not sure I like the idea of having a constructor like `new StreamGroupOptions(true)` because reading the code we don't know what true entails in terms of behaviour. We could have an enum or a constant that we set instead to make it explicit to the reader that it we are making a new stream. Something like `new StreamGroupOptions(MKSTREAM)`

Reply:
Hmm.. I think in this case builder is better unless the parameter is more descriptive but unfortunately it's just a boolean.